trunk/src/adapt/global_href.F90

```diff
@@ -18,14 +18,21 @@ subroutine global_href
 !..collect elements
    nr_elem = NRELES
 !
+! TODO: make this routine safe to use
+!       calling break directly can result in 2-irregular mesh temporarily
+!       calling refine fixes the issue but is much more expensive
+!       if breaking elements in the right order, it should be safe to do
+!
 !..break the elements
    do i=1,nr_elem
       mdle = ELEM_ORDER(i)
       if (is_leaf(mdle)) then
          call get_isoref(mdle, kref)
          call break(mdle,kref)
+!         call refine(mdle,kref)
```

Comment:
Why not just make this change at this point? Refine is only about 2x more expensive than break (it typically has 1 extra call to elem_nodes) and with the node_type change it really shouldn't be that much of a bottleneck

Reply:
Yes, I can make that change here, so it doesn't cause trouble in the future.